### PR TITLE
Remove `asfileobj=False` from a bunch of examples loading sample_data.

### PR DESCRIPTION
**Pull request by anntzer**
Using a file object works just fine.

There are some more examples that use genfromtxt, but their conversion
    effectively needs numpy>=1.14 per
    https://github.com/numpy/numpy/blob/master/doc/release/1.14.0-notes.rst#encoding-argument-for-text-io-functions

There are some more examples that load bivariate_normal.npy, but they
have a bunch of duplicated code that could use some more refactor.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/images_contours_and_fields/watermark_image.py [hunk 1]
```diff
@@ -5,25 +5,21 @@
...
-print('loading %s' % datafile)
-im = image.imread(datafile)
-im[:, :, -1] = 0.5  # set the alpha channel
+with cbook.get_sample_data('logo2.png') as file:
+    im = image.imread(file)
 
 fig, ax = plt.subplots()
 
-ax.plot(np.random.rand(20), '-o', ms=20, lw=2, alpha=0.7, mfc='orange')
+ax.plot(np.sin(10 * np.linspace(0, 1)), '-o', ms=20, alpha=0.7, mfc='orange')
```
Comment (anntzer) [inline examples/images_contours_and_fields/watermark_image.py]: the actual data here doesn't really matter, so let's not bother with seeding an rng.

File: examples/images_contours_and_fields/watermark_image.py [hunk 2]
```diff
@@ -5,25 +5,21 @@
...
 Using a PNG file as a watermark.
 """
+
 import numpy as np
 import matplotlib.cbook as cbook
 import matplotlib.image as image
 import matplotlib.pyplot as plt
 
-# Fixing random state for reproducibility
-np.random.seed(19680801)
```
Comment (tacaswell) [inline examples/images_contours_and_fields/watermark_image.py]: Can we please leave the seed?

File: examples/mplot3d/custom_shaded_3d_surface.py [hunk 3]
```diff
@@ -13,8 +13,8 @@
 import numpy as np
 
 # Load and format data
-filename = cbook.get_sample_data('jacksboro_fault_dem.npz', asfileobj=False)
-with np.load(filename) as dem:
+with cbook.get_sample_data('jacksboro_fault_dem.npz') as file, \
+     np.load(file) as dem:
```
Comment (jklymak) [inline examples/mplot3d/custom_shaded_3d_surface.py]: I find this hard to read, but not impossible.  Sure you don't want to just double-with it?

Comment (jklymak) [reply examples/mplot3d/custom_shaded_3d_surface.py]: @tacaswell approves so I guess its good...

**Pull request merged**